hooks/open-telemetry/src/main/java/dev/openfeature/contrib/hooks/otel/MetricHookOptions.java

```diff
@@ -0,0 +1,33 @@
+package dev.openfeature.contrib.hooks.otel;
+
+import dev.openfeature.sdk.ImmutableMetadata;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.opentelemetry.api.common.Attributes;
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * OpenTelemetry hook options.
+ */
+@Builder
+@Getter
+@SuppressFBWarnings(value = {"EI_EXPOSE_REP"}, justification = "Dimension list can be mutable")
+
+public class MetricHookOptions {
+
+    /**
+     * Custom handler to derive {@link Attributes} from flag evaluation metadata represented with
+     * {@link ImmutableMetadata}.
+     */
+    private Function<ImmutableMetadata, Attributes> attributeSetter;
+
+    /**
+     * List of {@link DimensionDescription} to be extracted from flag evaluation metadata.
+     */
+    @Builder.Default
+    private final List<DimensionDescription> setDimensions = Collections.emptyList();
```

Comment:
why do we need two different methods?

Reply:
This is to be backward compatible with the current release [^1] where we allow a list of Dimension descriptions to be extracted from metadata. 

With the proposed change of this PR, we are trying to provide call-back support aligning with OTel hook of JS SDK

[^1]: https://github.com/open-feature/java-sdk-contrib/blob/dev.openfeature.contrib.hooks.otel-v2.0.0/hooks/open-telemetry/src/main/java/dev/openfeature/contrib/hooks/otel/MetricsHook.java#L53-L59

Reply:
I see 👍  thanks for the explanation. Then, I'd suggest adding the `@Deprecated` annotation here to the old method 🤔

Reply:
Hmmm. The old method also converts the `DimensionDescription` to attributesBuilder from flagmeta data. I would prefer to expose one single method in SDK to create atrributes and not to confuse the user.

Reply:
@thiyagu06 yes that's correct. However, they can be considered as two approaches (one for shallow extraction and the other for customized extractions).

I am open to any opinion here. If we want to remove, then this should be done in two steps - first deprecate and then remove to avoid usage interruptions.

Reply:
Hmm. I agree. In that case, i would prefer to  keep the lambda based approach which provides both option of extracting attributes. Also, its one less concept for (`DimensionDescription` ) SDK users to understand the attribute builder for otel metrics.

Lets hear other thoughts as well.

Reply:
@thisthat @toddbaert @beeme1mr appreciate your feedback on this :)

Reply:
I think my slight preference is to keep just the one option, but I tend to agree with @Kavindu-Dodan that marking for deprecation is a better first step.

Honestly I don't feel very strongly either way on this one.

Reply:
@toddbaert thank you for the feedback

I added deprecation notice for the dimension description construct - https://github.com/open-feature/java-sdk-contrib/pull/391/commits/396ab41d5da3d036600d28606d6dd7c12376bfa5

We still have this through `MetricHookOptions` but eventually we can deprecate that too.